include/openthread/border_routing.h

```diff
@@ -218,6 +240,26 @@ void otBorderRoutingClearRouteInfoOptionPreference(otInstance *aInstance);
  */
 otError otBorderRoutingGetOmrPrefix(otInstance *aInstance, otIp6Prefix *aPrefix);
 
+/**
+ * Gets the PD provided off-mesh-routable (OMR) prefix.
+ *
+ * The prefix is extracted from the platform generated RA messages handled by
+ * otBorderRoutingProcessIcmp6Ra. When there are no valid platform provided OMR prefix, the returned
+ * prefix length will be 0.
```

Comment:
Can we use the `OT_ERROR_NOT_FOUND` to indicate the miss of PD OMR prefix. I want the output parameters always be valid when the error code is `OT_ERROR_NONE`, in which case is easier to use (i.e. client doesn't need to check `aPrefixInfo` again to ensure it succeeded).